utilities/pipelines/resourcePublish/Publish-ModuleToUniversalArtifactFeed.ps1

```diff
@@ -0,0 +1,118 @@
+#>
+function Publish-ModuleToUniversalArtifactFeed {
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory)]
+        [string] $TemplateFilePath,
+
+        [Parameter(Mandatory)]
+        [string] $VstsOrganizationUri,
+
+        [Parameter(Mandatory = $false)]
+        [string] $VstsFeedProject = '',
+
+        [Parameter(Mandatory)]
+        [string] $VstsFeedName,
+
+        [Parameter(Mandatory = $false)]
+        [string] $BearerToken = $env:TOKEN,
+
+        [Parameter(Mandatory)]
+        [string] $ModuleVersion
+    )
+
+    begin {
+        Write-Debug ('{0} entered' -f $MyInvocation.MyCommand)
+    }
+
+    process {
+
+        #################################
+        ##    Generate package name    ##
+        #################################
+
+        # Universal package names => lowercase alphanumerics, dashes, dots or underscores, under 256 characters.
+        # 'C:\arm\Microsoft.KeyVault\vaults\deploy.bicep' => 'microsoft.keyvault.vaults'
+        $ModuleFolderPath = Split-Path $TemplateFilePath -Parent
+        $universalPackageModuleName = $ModuleFolderPath.Replace('\', '/').Split('/arm/')[1]
+        $universalPackageModuleName = ($universalPackageModuleName.Replace('\', '.').Replace('/', '.').toLower() -Replace '[^a-z0-9\.\-_]')[0..255] -join ''
+        Write-Verbose "The universal package name is [$universalPackageModuleName]" -Verbose
+
+        ###########################
+        ##    Find feed scope    ##
+        ###########################
+        $feedScope = 'organization'
+
+        if (-not [string]::IsNullOrEmpty($VstsFeedProject)) {
+            $feedScope = 'project'
+        }
+        Write-Verbose "The package feed scope is [$feedScope]" -Verbose
+
+        #############################################
+        ##    Publish to Universal Package Feed    ##
+        #############################################
+        if ($PSCmdlet.ShouldProcess("Universal Package Feed entry [$universalPackageModuleName] version [$ModuleVersion] to feed [$VstsOrganizationUri/$VstsFeedProject/$VstsFeedName]", 'Publish')) {
+            $env:AZURE_DEVOPS_EXT_PAT = $BearerToken
+            $inputObject = @(
+                '--organization', "$VstsOrganizationUri",
+                '--feed', "$VstsFeedName",
+                '--scope', "$feedScope",
+                '--name', "$universalPackageModuleName",
+                '--version', "$ModuleVersion",
+                '--path', "$ModuleFolderPath",
+                '--description', "$universalPackageModuleName Module",
+                '--verbose'
+            )
+            if (-not [string]::IsNullOrEmpty($VstsFeedProject)) {
+                $inputObject += @('--project', "$VstsFeedProject")
+            }
+
+            Write-Verbose 'The command to publish to the feed is:' -Verbose
+            $command = "az artifacts universal publish $($inputObject -join ' ')"
+            Write-Verbose $command -Verbose
```

Comment:
is still really relevant to print at all times??

Reply:
Removed, aligning with templatespec and bicep reg approach.